enhancements/api-review/custom-tags-aws.md

```diff
@@ -48,62 +50,147 @@ Motivations include but are not limited to:
+    // See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    // AWS supports a maximum of 50 tags per resource. OpenShift reserves 25 tags for its use, leaving 25 tags
+    // available for the user.
+    // While ResourceTags field is mutable, items can not be removed.
+    // +kubebuilder:validation:MaxItems=25
+    // +optional
+    ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+}
+
+```
+
+```yaml
+spec:
+description: AWS contains settings specific to the Amazon Web Services Infrastructure provider.
+type: object
+properties:
+    resourceTags:
+    description: ResourceTags is a list of additional tags to apply to AWS resources created for the cluster. See https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for information on tagging AWS resources.
+    type: array
+    maxItems: 25
+    items:
+        description: AWSResourceTag is a tag to apply to AWS resources created for the cluster.
+        type: object
+        required:
+            - key
+            - value
+        properties:
+            key:
+                description: key is the key of the tag
+                type: string
+                maxLength: 128
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+            value:
+                description: value is the value of the tag. Some AWS service do not support empty values. Since tags are added to resources in many services, the length of the tag value must meet the requirements of all services.
+                type: string
+                maxLength: 256
+                minLength: 1
+                pattern: ^[0-9A-Za-z_.:/=+-@]+$
+```
+
 ### User Stories
 
 https://issues.redhat.com/browse/SDE-1146 - IAM users and roles can only operate on resources with specific tags
 As a security-conscious ROSA customer, I want to restrict the permissions granted to Red Hat in my AWS account by using
 AWS resource tags.
 
+### API Extensions
+
+This proposal edits `Infrastructure.config.openshift.io/v1` type
+
+### Operational Aspects of API Extensions
+
+NA
+
 ### Risks and Mitigations
 
+#### Failure Modes
+
+NA
+
+#### Support Procedures
+
+NA
+
 ## Design Details
 
+User can add tags during installation for creation of resources with tags. Installer creates infrastructure manifests with `resourceTags` to `.spec.platformSpec.aws` of the `Infrastructure.config.openshift.io`.
+Cluster config operator reconciles Infrastructure object, validates and updates `resourcesTags` from `.spec.platformSpec.aws` to  `.status.platformStatus.aws`.
+Operators that operate on aws resources must merge the tags from `.spec.platformSpec.aws`, `.status.platformStatus.aws` and individual component tags.
+
+The values from `status.platformStatus.aws` will used to only support downgrade.
+
+In case of precedence conflict or errors, the same will be reported using `Events` by the operators.
```

Comment:
how could those arise? If there's a precedence then the last-write always wins, so to say

Reply:
The precedence is to maintain creator/updator tool (in-case of external tool usage) remains same for user-defined tags which are created correspondingly. For example, the user tag is created during installation, can be updated by infrastructure CRD update. Any update using other CLI tools to update, will be overwritten on next reconcile cycle by the resource operator.